src/datanode/src/instance/sql.rs

```diff
@@ -106,18 +106,27 @@ impl Instance {
                 let name = alter_table.table_name().clone();
                 let (catalog, schema, table) = table_idents_to_full_name(&name, query_ctx.clone())?;
                 let table_ref = TableReference::full(&catalog, &schema, &table);
-                let req = SqlHandler::alter_to_request(alter_table, table_ref)?;
+                // Currently, we have to get the table multiple times. Consider remove the sql handler in the future.
```

Comment:
Maybe leave a todo here in case we forget.

Reply:
I created issue #1824 to track this.